packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -580,6 +568,33 @@ export class DatabaseCluster extends DatabaseClusterNew {
       target: this,
     });
   }
+
+  private addSingleUserRotationForMasterSecret(options: RotationSingleUserOptions): secretsmanager.SecretRotation {
+    if (!this.secret) {
+      throw new Error('Cannot add single user rotation for a cluster without master secret.');
+    }
+
+    const id = 'RotationSingleUser';
+    return this.addSingleUserRotationForSecret(this, id, this.secret, options);
+  }
+
+  private addSingleUserRotationForSecret(scope: Construct, id: string, secret: secretsmanager.ISecret, options: RotationSingleUserOptions)
+    : secretsmanager.SecretRotation {
+    if (Node.of(scope).tryFindChild(id)) {
+      const secretName = this === scope ? 'master' : 'additional'; // Is there a better way to do this?
+      throw new Error(`A single user rotation for ${secretName} secret was already added to this cluster.`);
+    }
+
```

Comment:
https://github.com/aws/aws-cdk/pull/13799#discussion_r604008157 Let me know if inconsistency is a concern.

Reply:
OK. Can you add a "ToDo" comment please referencing https://github.com/aws/aws-cdk/issues/13719?

Reply:
Hmmm, wait though. We _do_ call `attache` for `addRotationSingleUser()`, no? [Here](https://github.com/aws/aws-cdk/blob/246b393cccd6891aa0f3a74ead6953c3309927e3/packages/%40aws-cdk/aws-rds/lib/instance.ts#L1012-L1012).

So where's the inconsistency?

Reply:
That's an exceptional case IMO, because the user doesn't create the secret. Anyway, I meant `addRotationMultiUser` which doesn't attach the secret (explained in the other ticket for this auto-attach behavior).

Reply:
Speaking of, What's significant about getting the ID of construct from user? I'm asking because `addRotationMultiUser` is getting it from the caller, but for this one we generate it inside the method. I can go with an overloaded version of `addRotationSingleUser(id: string, options: RotationSingleUserOptions)`. Let me know what you think.

Reply:
> Speaking of, What's significant about getting the ID of construct from user? I'm asking because `addRotationMultiUser` is getting it from the caller, but for this one we generate it inside the method.

Again, this is a consequence of the fact that it was assumed `addRotationSingleUser()` would only ever be called once, while `addRotationMultiUser()` could be called multiple times.

>  I can go with an overloaded version of `addRotationSingleUser(id: string, options: RotationSingleUserOptions)`. Let me know what you think.

Unfortunately, that's a non-starter (TypeScript doesn't have overloads).

Reply:
> That's an exceptional case IMO, because the user doesn't create the secret. Anyway, I meant `addRotationMultiUser` which doesn't attach the secret (explained in the other ticket for this auto-attach behavior).

That's not true, right? Look at this:

```ts
    const secret = new DatabaseSecret(this, 'Secret', {
      username: 'admin',
      secretName: 'custom-secret-name',
    });

    new DatabaseCluster(this, 'Database', {
      credentials: Credentials.fromSecret(secret),
      // ...
    });
```

In this case, a Secret is created by the user, but `attach` is still called on it.

Reply:
> By default, the master password will be generated and stored in AWS Secrets Manager with auto-generated description.
Copied from the docs. In this situation the library has to attach the secret automatically otherwise it's gonna be confusing for the user. The fact that the library lets user to attach the secret doesn't change the situation with the default case. One solution could've been to detect if it's a user-defined secret or not, but that's a bit too much. The ideal solution would've been to do the attaching automatically for all cases (auto master, user-defined mater, as well as multi-user-rotated secrets).
Anyhow, I don't have strong opinion. Will add the attaching logic.

Reply:
As for the getting ID as an argument, that's a bit unfortunate. Not sure what's the best way to go. Let me know what's your call. BTW can we fix all these issues in V2?

Reply:
> As for the getting ID as an argument, that's a bit unfortunate. Not sure what's the best way to go. Let me know what's your call. BTW can we fix all these issues in V2?

The way to solve that is to add a new method instead of changing `addRotationSingleUser()`.

Reply:
Do you want me to do that? like `addAdditionalRotationSingleUser`?
What about V2?

Reply:
I like `addAdditionalRotationSingleUser()`.

> What about V2?

We try to minimize breaking changes between V1 and V2, to make the migration smoother.

Reply:
Will refactor accordingly.

Reply:
Thanks!